app/assets/javascripts/course/assessment/submission/answer/answer.js

```diff
@@ -85,6 +87,111 @@
   }
 
   /**
+   * Handles the answer comment edit button click event.
+   *
+   * @param e The event object.
+   */
+  function onCommentEdit(e) {
+    var $element = $(e.target);
+    var $post = $element.parents('.discussion_post:first');
+
+    var courseId = ANSWER_HELPERS.courseIdForElement($element);
+    var assessmentId = ANSWER_HELPERS.assessmentIdForElement($element);
+    var submissionId = ANSWER_HELPERS.submissionIdForElement($element);
+    var answerId = ANSWER_HELPERS.answerIdForElement($element);
+    var postId = $post.data('postId');
+    var postContent = $post.find('.content').text();
+    var postCommenter = $post.find('.user').html();
+
+    $post.children().hide();
+    var $form = createCommentForm($post, courseId, assessmentId, submissionId,
+                                  answerId, postId, postContent, postCommenter);
+    e.preventDefault();
+  }
+
+  /**
+   * Finds the form for a given answer comment.
+   *
+   * @param {jQuery} $element The answer comment to search for the form in.
+   * @return {jQuery} The answer comment form which was found.
+   */
+  function findAnswerCommentForm($element) {
+    return $element.find('> div.answer-comment-form');
+  }
+
+  /**
+   * Creates a form for the user to edit his answer comment.
+   *
+   * @param {jQuery} $element The element to search for the form.
+   * @param {Number} courseId The course ID that the comment is associated with.
+   * @param {Number} assessmentId The assessment ID that the comment is associated with.
+   * @param {Number} submissionId The submission ID that the comment is associated with.
+   * @param {Number} answerId The answer ID that the comment is associated with.
+   * @param {Number} postId The ID of the comment being edited.
+   * @param {String} postContent The existing contents of the comment's text field.
+   * @param {HTMLElement} postCommenter The commenter's name with a link to his profile.
+   * @return {jQuery} The form for the annotation post.
+   */
+  function createCommentForm($element, courseId, assessmentId, submissionId, answerId,
+                             postId, postContent, postCommenter) {
+    $element.append(render('comment_form', {
+      courseId: courseId,
+      assessmentId: assessmentId,
+      submissionId: submissionId,
+      answerId: answerId,
+      postId: postId,
+      postContent: postContent,
+      postCommenter: postCommenter
+    }));
+
+    return findAnswerCommentForm($element);
+  }
+
+  /**
+   * Handles the reset of the answer comment form.
+   *
+   * @param e The event object.
+   */
+  function onCommentFormResetted(e) {
```

Comment:
Can we use `onCommentFormReset`? (Past tense of reset is reset)

The other function can be `onCommentFormSubmit` or `onCommentFormSubmission`. Using Submitted sounds more like a callback.